fl4health/strategies/feddg_ga_strategy.py

```diff
@@ -0,0 +1,382 @@
+
+        # Setting self.num_rounds
+        if self.num_rounds is None:
+            assert self.on_fit_config_fn is not None, "on_fit_config_in must be specified"
+            config = self.on_fit_config_fn(server_round)
+            assert "n_server_rounds" in config, "n_server_rounds must be specified"
+            assert isinstance(config["n_server_rounds"], int), "n_server_rounds is not an integer"
+            self.num_rounds = config["n_server_rounds"]
+
+        return client_fit_ins
+
+    def aggregate_fit(
+        self,
+        server_round: int,
+        results: List[Tuple[ClientProxy, FitRes]],
+        failures: List[Union[Tuple[ClientProxy, FitRes], BaseException]],
+    ) -> Tuple[Optional[Parameters], Dict[str, Scalar]]:
+        """
+        Aggregate fit results by weighing them against the adjustment weights and then summing them.
+
+        Collects the fit metrics that will be used to change the adjustment weights for the next round.
+
+        Args:
+            server_round: (int) the current server round.
+            results: (List[Tuple[ClientProxy, FitRes]]) The clients' fit results.
+            failures: (List[Union[Tuple[ClientProxy, FitRes], BaseException]]) the clients' fit failures.
+
+        Returns:
+            (Tuple[Optional[Parameters], Dict[str, Scalar]]) A tuple containing the aggregated parameters
+                and the aggregated fit metrics.
+        """
+
+        _, metrics_aggregated = super().aggregate_fit(server_round, results, failures)
+
+        self.train_metrics = {}
+        for result in results:
+            self.train_metrics[result[0].cid] = result[1].metrics
+
+        parameters_aggregated = ndarrays_to_parameters(self.weight_and_aggregate_results(results))
+
+        return parameters_aggregated, metrics_aggregated
+
+    def aggregate_evaluate(
+        self,
+        server_round: int,
+        results: List[Tuple[ClientProxy, EvaluateRes]],
+        failures: List[Union[Tuple[ClientProxy, EvaluateRes], BaseException]],
+    ) -> Tuple[Optional[float], Dict[str, Scalar]]:
+        """
+        Aggregate evaluation losses using weighted average.
+
+        Collects the evaluation metrics and update the adjustment weights, which will be used
+        when aggregating the results for the next round.
+
+        Args:
+            server_round: (int) the current server round.
+            results: (List[Tuple[ClientProxy, FitRes]]) The clients' evaluate results.
+            failures: (List[Union[Tuple[ClientProxy, FitRes], BaseException]]) the clients' evaluate failures.
+
+        Returns:
+            (Tuple[Optional[float], Dict[str, Scalar]]) A tuple containing the aggregated evaluation loss
+                and the aggregated evaluation metrics.
+        """
+        loss_aggregated, metrics_aggregated = super().aggregate_evaluate(server_round, results, failures)
+
+        self.evaluation_metrics = {}
+        for result in results:
```

Comment:
Super minor, but since results is a tuple, it might be slightly easier to read if we unpack the tuple here explicitly. That is, 
```
for client_proxy, eval_res in results:
....
```